service-base/src/main/java/org/eclipse/hono/service/auth/device/UsernamePasswordCredentials.java

```diff
@@ -114,27 +115,31 @@ public final String getPassword() {
     @Override
     public boolean matchesCredentials(final JsonObject candidateSecret) {
 
+        encoder = PasswordEncoderFactory.createDelegatingPasswordEncoder();
+
         try {
             final String pwdHash = candidateSecret.getString(CredentialsConstants.FIELD_SECRETS_PWD_HASH);
             if (pwdHash == null) {
                 LOG.debug("candidate hashed-password secret does not contain a pwd hash");
                 return false;
             }
 
-            final byte[] hashedPasswordOnRecord = Base64.getDecoder().decode(pwdHash);
+            final String hashFunction = candidateSecret.getString(
+                    CredentialsConstants.FIELD_SECRETS_HASH_FUNCTION,
+                    CredentialsConstants.DEFAULT_HASH_FUNCTION);
+
+            String encodedPassword = "{" + hashFunction + "}";
```

Comment:
I think in this case it would be advisable to use a `StringBuilder` instead of concatenating strings.